Ad7192 upstream iio v2

mirceacaprioru opened the pull request:
Upstreaming the last modifications, hopefully, to get this driver out of staging with all the required features like: 
- set filter frequency
- channel calibration

commodo commented on drivers/iio/adc/ad_sigma_delta.c:
```diff
@@ -205,7 +205,7 @@ int ad_sd_reset(struct ad_sigma_delta *sigma_delta,
 EXPORT_SYMBOL_GPL(ad_sd_reset);
 
-static int ad_sd_calibrate(struct ad_sigma_delta *sigma_delta,
+int ad_sd_calibrate(struct ad_sigma_delta *sigma_delta,
```
the title here can be a bit confusing;
this patch is essentially exporting the `ad_sd_calibrate` ; so we should probably emphasize on that aspect more, than on the aspect of something being removed [as if un-needed]

mirceacaprioru replied:
I agree, it does look like something else from the commit message.

commodo commented on drivers/staging/iio/adc/ad7192.c:
```diff
@@ -169,10 +169,56 @@ struct ad7192_state {
+
+	st->syscalib_mode[chan->channel] = mode;
+
+	if (!strcmp(ad7192_syscalib_modes[mode], "zero_scale"))
```
maybe do here an enum 
```
enum {
   AD7192_SYSCALIB_ZERO_SCALE,
   AD7192_SYSCALIB_FULL_SCALE,
};

and then
```
static const char * const ad7192_syscalib_modes[] = {
	[AD7192_SYSCALIB_ZERO_SCALE] = "zero_scale",
	[AD7192_SYSCALIB_FULL_SCALE] = "full_scale",
};
```

the whole point of iio_enum, seems to be, that it converts string to int for you;
so, here you should get:
```
	if (mode == AD7192_SYSCALIB_ZERO_SCALE)
```

commodo commented on drivers/staging/iio/adc/ad7192.c:
```diff
@@ -713,12 +705,10 @@ static int ad7192_probe(struct spi_device *spi)
 		st->int_vref_mv = voltage_uv / 1000;
 	else
-		dev_warn(&spi->dev, "reference voltage undefined\n");
+		dev_err(&spi->dev, "Device tree error, reference voltage undefined\n");
```
just a note: this patch can be a separate patch when sending upstream;
you don't need to make it part of any series;
and it looks like it could apply without the other patches

commodo commented on Documentation/devicetree/bindings/iio/adc/adi,ad7192.yaml:
```diff
@@ -0,0 +1,122 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
```
hmm; i think this would be GPL-2.0   or ?

mirceacaprioru replied:
This is in concordance with a review I got from Rob on the ad7124 yaml binding. He requested it to be this format.

commodo replied:
ack

commodo commented on Documentation/devicetree/bindings/iio/adc/adi,ad7192.yaml:
```diff
@@ -0,0 +1,122 @@
+
+examples:
+  - |
+    ad7192@0 {
```
this should probably be wrapped in an spi block;
something like
```
spi0 {
   ad7192@0 {
        ....
    };
};
```

mirceacaprioru replied:
ack

commodo commented on Documentation/devicetree/bindings/iio/adc/adi,ad7192.yaml:
```diff
@@ -0,0 +1,122 @@
+    description: SPI chip select number for the device
+    maxItems: 1
+
+  spi-cpol:
```
is this required?
if so, it's also just   `spi-cpol: true` 
same for `spi-cpha`

commodo commented on Documentation/devicetree/bindings/iio/adc/adi,ad7192.yaml:
```diff
@@ -0,0 +1,122 @@
+    ad7192@0 {
+    	compatible = "adi,ad7192";
+    	reg = <0>;
+    	spi-max-frequency = <1000000>;
```
`spi-max-frequency` is usually not placed in examples;
but i guess it's fine to leave it

nunojsa commented on drivers/staging/iio/adc/ad7192.c:
```diff
@@ -409,6 +401,49 @@ static ssize_t ad7192_set(struct device *dev,
+	freq[2] = DIV_ROUND_CLOSEST(fadc * 230, 1024);
+	freq[3] = DIV_ROUND_CLOSEST(fadc * 272, 1024);
+}
+
```
nitpick: Do we really want the `inline`? I would leave that decision to the compiler :smile:.
Take a look at [inline-disease](https://www.kernel.org/doc/html/v4.10/process/coding-style.html#the-inline-disease).

mirceacaprioru replied:
I think i copy-pasted it from somewhere, will remove

mirceacaprioru commented:
Changelog  v1 -> v2:
- removed inline 
- set spi-cpol/cpha to true and added to required properties
- added spi block wrapper
- added enum for zero_scale, full_scale calibration
- changed commit message for ad_sd_calibrate export

commodo commented:
fine from my side to close and send it upstream;

mirceacaprioru commented:
ok, thanks for the review.
